### PR TITLE
feat(user_management): grade_9_mode

### LINKED ISSUE #56: Feature: Grade 9
Under Miscellaneous:

"Grade_9_Mode"

To toggle on/off:

- [ ] Entrance Theme player (set volume to 0?)
- [ ] TV # 1-3

### PR DESCRIPTION
**Pull request by punctuations**
added grade_9_mode command which will close #56

### CODE REVIEW

File: scripts/User_Management/grade_9_mode.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+from scripts._utils import ssh, pi, utils
+
+host_themes = "pi-themes"
+tvs = ["1", "2", "3", "4"]
```
Comment (tylerecouture) [inline scripts/User_Management/grade_9_mode.py:4]: Don't redefine/duplicate code.  keep it dry and import these from the source.

File: scripts/User_Management/grade_9_mode.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+from scripts._utils import ssh, pi, utils
```
Comment (tylerecouture) [inline scripts/User_Management/grade_9_mode.py:1]: Per issue, this script should be in Miscellaneous

File: scripts/User_Management/grade_9_mode.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+host_themes = "pi-themes"
+tvs = ["1", "2", "3", "4"]
+
+TV_OFF_CMD = "echo standby 0 | cec-client -s -d 1"
+TV_ON_CMD = "echo on 0 | cec-client -s -d 1"
```
Comment (tylerecouture) [inline scripts/User_Management/grade_9_mode.py:7]: import from elsewhere, don't duplicate.

File: scripts/User_Management/grade_9_mode.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+def engage():
+    ssh_connection = ssh.SSH(host_themes, pi.username, pi.password)
+    ssh_connection.send_cmd("amixer sset 'Headphone',0 0%")
+
+    for tv_number in tvs:
```
Comment (tylerecouture) [inline scripts/User_Management/grade_9_mode.py:14]: don't duplicate, call the method from elsewhere.  If it isn't a method, make it one so code can be DRY

File: scripts/User_Management/grade_9_mode.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+def disengage():
+    ssh_connection = ssh.SSH(host_themes, pi.username, pi.password)
+    ssh_connection.send_cmd("amixer sset 'Headphone',0 100%")
+
+    for tv_number in tvs:
+        hostname = "pi-tv{}".format(tv_number)
+        ssh_connection = ssh.SSH(hostname, pi.username, pi.password)
+
+        ssh_connection.send_cmd(TV_ON_CMD)
+        ssh_connection.close()
```
Comment (tylerecouture) [inline scripts/User_Management/grade_9_mode.py:31]: ditto

File: scripts/User_Management/grade_9_mode.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+
+def grade_9_mode():
+    confirmation = utils.input_styled("Do you want to engage grade 9s? [y/N] (q to quit): \n")
```
Comment (tylerecouture) [inline scripts/User_Management/grade_9_mode.py:35]: This is ambiguous, and it's not clear from prompt the difference between N and q. Should ask to [e]ngage, [D]isengage, or [q]uit.

Should also have a description of what it does:
" Enganging grade 9 mode will:
   * Thing 1
   * Thing 2
   * etc.
"

**Review by tylerecouture — CHANGES_REQUESTED** [review]
a few tweaks, thank you!

File: scripts/User_Management/grade_9_mode.py (hunk 6 above)
Comment (punctuations) [reply scripts/User_Management/grade_9_mode.py]: fixed as per [6612eae](https://github.com/timberline-secondary/hackerspace-scripts-2/pull/60/commits/6612eae166341b92fd6439632f661f323301261d) and [8bec1cf](https://github.com/timberline-secondary/hackerspace-scripts-2/pull/60/commits/8bec1cfe789ef129a71e89d962594cdf4165f7fe)

File: scripts/User_Management/grade_9_mode.py (hunk 5 above)
Comment (punctuations) [reply scripts/User_Management/grade_9_mode.py]: fixed as per [6612eae](https://github.com/timberline-secondary/hackerspace-scripts-2/pull/60/commits/6612eae166341b92fd6439632f661f323301261d)

File: scripts/User_Management/grade_9_mode.py (hunk 4 above)
Comment (punctuations) [reply scripts/User_Management/grade_9_mode.py]: fixed as per [6612eae](https://github.com/timberline-secondary/hackerspace-scripts-2/pull/60/commits/6612eae166341b92fd6439632f661f323301261d)

File: scripts/User_Management/grade_9_mode.py (hunk 2 above)
Comment (punctuations) [reply scripts/User_Management/grade_9_mode.py]: fixed as per [6612eae](https://github.com/timberline-secondary/hackerspace-scripts-2/pull/60/commits/6612eae166341b92fd6439632f661f323301261d)

File: scripts/User_Management/grade_9_mode.py (hunk 3 above)
Comment (punctuations) [reply scripts/User_Management/grade_9_mode.py]: im confused

File: scripts/User_Management/grade_9_mode.py (hunk 1 above)
Comment (punctuations) [reply scripts/User_Management/grade_9_mode.py]: im confused

**Comment by punctuations** [comment]
I will update the for loop to be threaded so it all shuts down at once, instead of slowly one-by-one!

File: scripts/User_Management/grade_9_mode.py (hunk 3 above)
Comment (tylerecouture) [reply scripts/User_Management/grade_9_mode.py]: These commands already exist in another module, do they not? Don't repeat yourself, especially with hard coded commands like this. We need to update them, we'll update in one place and forget about the other.

Comment (tylerecouture) [reply scripts/User_Management/grade_9_mode.py]: Instead of copy and pasting this code from the other module, import the variables/constants.

Comment (punctuations) [reply scripts/User_Management/grade_9_mode.py]: they appear in one other command, should I move them to `TV/_utils.py` ?

Comment (punctuations) [reply scripts/User_Management/grade_9_mode.py]: solved as per [8bec1cf](https://github.com/timberline-secondary/hackerspace-scripts-2/pull/60/commits/8bec1cfe789ef129a71e89d962594cdf4165f7fe)

Comment (tylerecouture) [reply scripts/User_Management/grade_9_mode.py]: Yes please.

**Pull request merged**